CMakeLists.txt

```diff
@@ -587,7 +587,7 @@ if(WITH_OPTIM)
         endif()
         list(APPEND ZLIB_ARCH_HDRS ${ARCHDIR}/arm_features.h)
         list(APPEND ZLIB_ARCH_SRCS ${ARCHDIR}/arm_features.c)
-        if(WITH_ACLE AND NOT "${ARCH}" MATCHES "armv[2-7]")
+        if(WITH_ACLE AND NOT (MSVC AND "${ARCH}" MATCHES "arm") AND NOT "${ARCH}" MATCHES "armv[2-7]")
```

Comment:
Why not add the check into `check_acle_compiler_flag` instead?

Reply:
I don't think we should even be having `AND NOT "${ARCH}" MATCHES "armv[2-7]"` on this if statement. We can let the compiler to run the check for the flag. Because otherwise the `WITH_ACLE` flag is never turned off when printing used features.

Reply:
The check inside of `check_acle_compiler_flag` can use `if(MSVC AND MSVC_C_ARCHITECTURE_ID STREQUAL "ARM64")`

Reply:
> Do you still think it should be inside check_acle_compiler_flag?

_I_ think so and it would be a good to add a comment something like to that if statement in `check_acle_compiler_flag` like this:
```
# Missing crc32b instructions in Windows intrinsics headers for ARM 32-bit.
```

Reply:
> "ACLE" is a bit of a misnomer on msvc, but it's still the term zlib-ng uses to detect support for intrinsics.

I agree, we could address that separately when we end up using more instructions.

Reply:
Looks good. No need for the outer `else()`. You return early instead:
```
if(MSVC AND MSVC_C_ARCHITECTURE_ID STREQUAL "ARM64")
    set(HAVE_ACLE_FLAG TRUE)
    return()
endif()
```

Reply:
In that case it looks ok to me.